docker/pypi/wmagent/install.sh

```diff
@@ -0,0 +1,179 @@
+echo "-----------------------------------------------------------------------"
+echo "Start $stepMsg"
+
+# First upgrade pip to the latest version:
+pip install wheel
+pip install --upgrade pip
+
+# Second deploy the package. Interrupt on error:
+pip install wmagent==$WMA_TAG || { err=$?; echo "Failed to install wmagent:$WMA_TAG at $WMA_DEPLOY_IR" ; exit $err ; }
+echo "Done $stepMsg!" && echo
+echo "-----------------------------------------------------------------------"
+
+# Setup required directories
+stepMsg="Creating required directory structure in the WMAgent image"
+echo "-----------------------------------------------------------------------"
+echo "Start $stepMsg"
+mkdir -p ${WMA_DEPLOY_DIR} || true
+mkdir -p $WMA_BASE_DIR/wmagent/$WMA_TAG || true
+ln -s $WMA_BASE_DIR/wmagent/$WMA_TAG $WMA_CURRENT_DIR
+
+mkdir -p $WMA_ADMIN_DIR $WMA_HOSTADMIN_DIR $WMA_CERTS_DIR $WMA_MANAGE_DIR $WMA_INSTALL_DIR
+chmod 755 $WMA_CERTS_DIR
+
+cd $WMA_BASE_DIR
+echo "Done $stepMsg!" && echo
+echo "-----------------------------------------------------------------------"
+
+stepMsg="Downloading all files required for the containder intialisation at the host"
+echo "-----------------------------------------------------------------------"
+echo "Start $stepMsg"
+
+# Fix for outdated yui library - A really bad workaround. We should get rid of it ASAP:
+wget -nv -P $WMA_DEPLOY_DIR wget http://cmsrep.cern.ch/cmssw/repos/comp/slc7_amd64_gcc630/0000000000000000000000000000000000000000000000000000000000000000/RPMS/cd/cda5f9ef4b33696e67c9e2f995dd5cb6/external+yui+2.9.0-1-1.slc7_amd64_gcc630.rpm
+mkdir $WMA_DEPLOY_DIR/yui && cat $WMA_DEPLOY_DIR/external+yui+2.9.0-1-1.slc7_amd64_gcc630.rpm|rpm2archive - |tar --strip-components=13 -xzv --directory $WMA_DEPLOY_DIR/yui
+
+echo "Done $stepMsg!" && echo
+echo "-----------------------------------------------------------------------"
+
+stepMsg="Generating and preserving current build id"
+echo "-----------------------------------------------------------------------"
+echo "Start $stepMsg"
+echo $RANDOM |sha256sum |awk '{print $1}' > $WMA_ROOT_DIR/.dockerBuildId
+echo "WMA_BUILD_ID:`cat $WMA_ROOT_DIR/.dockerBuildId`"
+echo "WMA_BUILD_ID preserved at: $WMA_ROOT_DIR/.dockerBuildId "
+echo "Done $stepMsg!" && echo
+echo "-----------------------------------------------------------------------"
+
+stepMsg="Replace the current /data/manage script coming from 'dmwm-base' image with a symlink link"
+echo "-----------------------------------------------------------------------"
+echo "Start $stepMsg"
+[[ -f /data/manage ]] && rm -f /data/manage && ln -s $WMA_MANAGE_DIR/manage /data/manage
+echo "Done $stepMsg!" && echo
+echo "-----------------------------------------------------------------------"
+
+stepMsg="Creating all runtime aliases for the WMA_USER"
+echo "-----------------------------------------------------------------------"
+echo "Start $stepMsg"
+set -x
+cat <<EOF >> /home/${WMA_USER}/.bashrc
```

Comment:
How is that different from what I am doing here? Except that we  should juggle with one more temporary file. In the current implementation you do see the content on the screen, right here the next few lines. And it is concatenated to the original/standard `.bashrc`  without overwriting it.

Reply:
Ok then, in order to minimize the number of files/places where we deal with the environment, I'd like to reuse the already existing `env.sh` file and just add to it.  I'll also apply some basic tweaks to relate all the commands executed to the `WMA_ROOT_DIR`. That way, on a later stage, when we completely move away from the old deployment model and we no longer care about backwards compatibility, it will be good to have all the env tweaks in a single place  which would help us in the rewriting the deployment  scripts at https://github.com/dmwm/WMCore/tree/master/deploy.